_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_splitter.cpp

```diff
@@ -0,0 +1,259 @@
+            if (unitStart) // Start code was found. Need to close the cached unit
+            {
+                readSize = unitStart - begin; // Up to found startcode
+                m_cache.insert(m_cache.end(), (uint8_t *)begin, (uint8_t *)begin + readSize); // load data to the cache
+
+                unit = { m_cache.data(), m_cache.data() + m_cache.size(), m_cache[prefix_size], m_pts } ; // Construct unit from cached data
+                m_needCleanCache = true; // need to clean up the cache on next search iteration
+            }
+            else // start code wasn't found
+            {
+                readSize = end - begin;
+                m_cache.insert(m_cache.end(), (uint8_t *)begin, (uint8_t *)end); // load the whole data to the cache
+            }
+
+            in->MoveDataPointer(readSize);
+            return unit;
+        }
+
+        RawUnit naluTmp = FindRawUnit(begin, end);
+        if (-1 == naluTmp.type || (!naluTmp.begin && !naluTmp.end)) // nothing found
+        {
+            readSize = end - begin; // Skip all data
+        }
+        else if (naluTmp.begin && !naluTmp.end) // Found only where unit starts
+        {
+            // Consume all data from buffer except last 2 bytes.
+            // An exception from this is 'MFX_BITSTREAM_COMPLETE_FRAME' case.
+            // These two bytes are to prevent start code chopping between two separete FindRawUnit() calls.
+            uint32_t numBytesToLeave = !((m_source->GetFlags() & UMC::MediaData::FLAG_VIDEO_DATA_NOT_FULL_FRAME)) ? 0 : (prefix_size - 1);
+            readSize = end - begin - numBytesToLeave;
+            m_cache.insert(m_cache.end(), naluTmp.begin, end - numBytesToLeave); // But save to the cache only since start code
+            m_pts = in->GetTime();
+        }
+        else // found begin and end
+        {
+            readSize = naluTmp.end - begin; // Up to the end of the unit
+            unit = naluTmp;
+            unit.pts = in->GetTime();
+        }
+
+        in->MoveDataPointer(readSize);
+
+        return unit;
+    }
+
+    RawUnit RawHeaderIterator::OnEOS()
+    {
+        RawUnit nalu;
+        if (m_needCleanCache)
+        {
+            m_cache.clear();
+            m_needCleanCache = false;
+            m_pts = -1;
+        }
+
+        if (m_cache.size())
+        {
+            nalu = { m_cache.data(), m_cache.data() + m_cache.size(), m_cache[prefix_size], m_pts } ; // Construct unit from cached data
+            m_needCleanCache = true; // Need to clean up the cache on next iteration
+        }
+
+        return nalu;
+    }
+
+    uint8_t * RawHeaderIterator::FindStartCode(uint8_t * begin, uint8_t * end)
+    {
+        for (; begin < end - 3; ++begin)
+        {
+            if (begin[0] == 0 && begin[1] == 0 && begin[2] == 1)
+            {
+                return begin;
+            }
+        }
+        return nullptr;
+    }
+
+    RawUnit RawHeaderIterator::FindRawUnit(mfxU8 * begin, mfxU8 * end)
```

Comment:
Different types - _uint8_t_ in _FindStartCode_ but _mfxU8_ here. What is intention of that?

Reply:
No special intention. If you want, I can replace with uint8_t

Reply:
Ok